lib/tile_bag.rb

```diff
@@ -0,0 +1,37 @@
+require_relative 'scoring'
+
+module Scrabble
+  class TileBag
+
+    attr_reader :tiles
+
+    def initialize
+      @tiles = {:A => 9, :B => 2, :C => 2, :D => 4, :E => 12, :F => 2, :G => 3, :H => 2, :I => 9, :J => 1, :K => 1, :L => 4, :M => 2, :N => 6, :O => 8, :P => 2, :Q => 1, :R => 6, :S => 4, :T => 6, :U => 4, :V => 2, :W => 2, :X => 1, :Y => 2, :Z => 1}
+    end #end of initialize
+
+    def draw_tiles(num)
+      available_tiles = @tiles.select { |key, value| value != 0 }
+
+      if tiles_remaining > 0
+        drawn_tiles = available_tiles.keys.sample(num)
```

Comment:
I think this approach might have a subtle issue that impacts the randomness by which the tiles are drawn.

Specifically, it looks like it only ever draws tiles with distinct letters -- e.g. it would never draw multiple E's from the bag. There's a test that we could write for this which I think would expose the failing behavior (and thus allow you to verify that it has been corrected after making changes):

```ruby
it "should allow drawing duplicate letters" do
  bag = Scrabble::TileBag.new
  drawn_tiles = bag.draw_tiles(27)
  drawn_tiles.length.must_equal 27
end
```